tools/repository_manifest.py

```
@@ -0,0 +1,73 @@
+            "uid": metadata.st_uid,
+        }
+        if path.is_symlink():
+            target = os.readlink(path)
+            if "/" in target or target in ("", ".", ".."):
+                raise SystemExit(f"unsafe repository symlink: {path.name} -> {target}")
+            try:
+                target_metadata = (root / target).lstat()
+            except FileNotFoundError:
+                raise SystemExit(
+                    f"repository symlink target is missing: {path.name} -> {target}"
+                )
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)